core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -582,13 +586,16 @@ private[log] class Cleaner(val id: Int,
         transactionMetadata.addAbortedTransactions(abortedTransactions)
 
         val retainDeletesAndTxnMarkers = currentSegment.lastModified > deleteHorizonMs
-        info(s"Cleaning $currentSegment in log ${log.name} into ${cleaned.baseOffset} " +
+        info(s"Cleaning $currentSegment in log ${log.name} into ${currentSegment.baseOffset} " +
```

Comment:
This log message no longer makes sense. Is the intent to use `cleanedSegment` instead of `currentSegment`? Perhaps we should consider moving this log line to after the cleaning when we know what the new segment is.

Reply:
Hmmm I think I didn't really understand this log at first. It does make sense to move it to when we know what the new segment is. I think the issue I had was what to do with `retainDeletesAndTxnMarkers`. Maybe it makes sense to have a log saying the current segment and the deletion info and a second log to say what the new segment's baseOffset is later.

Reply:
Not sure I follow what the complication is. We can still mention whether deletes/markers have been retained after the fact. It would be useful to clearly indicate when no records were retained from a segment though.

Reply:
That makes sense.

Reply:
I think I was concerned about putting this information in the try block and it not printing. But I looking at it again, it's not a huge deal not to log in that case.